File: easy/ts/283.ts
// https://leetcode.com/problems/move-zeroes/description/
// Time Complexity: O(n)
// Space Complexity: O(1)
function moveZeroes(nums: number[]): void {
   let currentIndexToChange = 0
   for(let i = 0; i < nums.length; i++) {
       if(nums[i] !== 0) {
           nums[currentIndexToChange] = nums[i]
           currentIndexToChange++
       }
   }

   while(currentIndexToChange < nums.length) {
       nums[currentIndexToChange] = 0
       currentIndexToChange++
   }
};

File: easy/python/283.py
# https://leetcode.com/problems/move-zeroes/?envType=study-plan-v2&envId=leetcode-75
# Time: O(n)
# Space: O(1)
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        totalNonZeros = 0

        for element in nums:
            if element != 0:
                nums[totalNonZeros] = element
                totalNonZeros += 1

        while totalNonZeros < len(nums):
            nums[totalNonZeros] = 0
            totalNonZeros += 1
